packages/dev/core/src/Culling/ray.ts

```diff
@@ -798,7 +804,8 @@ Scene.prototype._internalPick = function (
             continue;
         }
 
-        const world = mesh.getWorldMatrix();
+        const forceCompute: boolean = !!(mesh.billboardMode !== AbstractMesh.BILLBOARDMODE_NONE && this.activeCameras && this.activeCameras.length > 1);
+        const world = mesh.computeWorldMatrix(forceCompute);

```

Comment:
I think the real bug is whithin the `Node.getWorldMatrix / TransformNode.computeWorldMatrix` functions, which should factor the camera that has been used to compute the last world matrix when deciding if the current cached result is ok or not. However, it should do so only if the computation depends on the camera (some paths in the computation don't need the camera).

That way, you don't need to add the lines above and `const world = mesh.getWorldMatrix()` is still ok.

It could be something like (not fully tested, but it works with the PG):

**Node class:**

```typescript
protected _cameraForComputeWorldMatrix: Nullable<Camera> = null;

public getWorldMatrix(): Matrix {
    if (this._currentRenderId !== this._scene.getRenderId() || this._cameraForComputeWorldMatrix !== this._scene.activeCamera) {
        this.computeWorldMatrix();
    }
    return this._worldMatrix;
}
```

**TransformNode class**:
```typescript
public computeWorldMatrix(force?: boolean): Matrix {
    if (this._isWorldMatrixFrozen && !this._isDirty) {
        return this._worldMatrix;
    }

    const camera = <Camera>this.getScene().activeCamera;
    const useBillboardPath = this._billboardMode !== TransformNode.BILLBOARDMODE_NONE && !this.preserveParentRotationForBillboard;
    const worldMatrixDependsOnCamera = camera && ((this._infiniteDistance && !this.parent) || (useBillboardPath && this.billboardMode));

    const currentRenderId = this.getScene().getRenderId();
    if ((!worldMatrixDependsOnCamera || this._cameraForComputeWorldMatrix === this._scene.activeCamera) && !this._isDirty && !force && (this._currentRenderId === currentRenderId || this.isSynchronized())) {
        this._currentRenderId = currentRenderId;
        return this._worldMatrix;
    }

    this._cameraForComputeWorldMatrix = worldMatrixDependsOnCamera ? camera : null;

    const useBillboardPosition = (this._billboardMode & TransformNode.BILLBOARDMODE_USE_POSITION) !== 0;

    this._updateCache();
    [same existing code]
```